### DISCUSSION TITLE
front_edge_offset_of_steppable_regionの用途

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#63] [Q&A] [answered]
`front_edge_offset_of_steppable_region`は、現在使われていますか.  また、どのような状況で使いますか.
https://github.com/YutaKojio/hrpsys-base/blob/49aeca84efde2d52655e3ee6b0fb0fc1219861cb/rtc/AutoBalancer/GaitGenerator.cpp#L1552-L1554

ここで使われていた形跡がありました。
https://github.com/YutaKojio/rtmros_tutorials/blob/88bf75387fd9c8408e217574376e4b393965d6d0/hrpsys_ros_bridge_tutorials/src/hrpsys_ros_bridge_tutorials/urata_hrpsys_config.py#L469

**Comment by YutaKojio** [comment, marked as answer]
visionがちゃんとしてる今はいらないはずです．
大森着地可能領域時代は，(i)段差上り時に足をひっかける，もしくは(ii)段差下り直前に踏み外すという失敗が多く，その2パターンを避ける方向に着地可能領域を仮想的にオフセットさせる目的で使ってました．

しかし，厳密には，この実装では参照着地位置が着地可能領域制限により打ち切られたときにオフセットされるので，参照着地位置がもともと着地可能領域内の段差ぎりぎりにある場合は改善されません．
